### PR TITLE
Holidays

### PR DESCRIPTION
**Pull request by FREDOCAKA**

### CODE REVIEW

File: app/Api/ApitransformCollection.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+namespace App\Api;
+class ApitransformCollection
```
Comment (coreysan) [inline app/Api/ApitransformCollection.php]: class casing: Api*T*ransformCollection

File: database/migrations/2017_03_09_085400_create_holidays_table.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
```
Comment (coreysan) [inline database/migrations/2017_03_09_085400_create_holidays_table.php]: This file is space-perfect!!! Jeffrey would be proud 👍

File: database/seeds/HolidaysSeeder.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            'name' => ' Martyr\'s Day',
+            'sacco_id' =>1,
+            'date' =>'2017-06-03'
+        ]);
+
+        Holiday::create([
+            'name' => 'Labour Day',
+            'sacco_id' =>2,
+            'date' =>'2017-05-01'
+        ]);
```
Comment (coreysan) [inline database/seeds/HolidaysSeeder.php]: Test your factory by creating a bunch of holidays at this stage.

File: resources/views/holidays/edit.blade.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+@extends('layouts.app')
+
+@section('content')
+
+  @component('shared/components/heading-and-buttons')
+    @slot('title')
+    <i class="mobis-i-home"></i>
```
Comment (coreysan) [inline resources/views/holidays/edit.blade.php]: indenting inside the slot

File: resources/views/holidays/index.blade.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
+@extends('layouts.app')
+
+@section('body-class','holidays')
+
```
Comment (coreysan) [inline resources/views/holidays/index.blade.php]: Add a `@section('title', 'Holidays')` here

File: resources/views/holidays/index.blade.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+@section('content')
+        
+ @component('shared/components/heading-and-buttons')
+    @slot('title')<i class="mobis-i-home"></i>
+            Holidays
+     @endslot
+     <a href="{{url("/holidays/create")}}" class="btn btn-success btn-group-custom">
```
Comment (coreysan) [inline resources/views/holidays/index.blade.php]: Some minor indentation issues here

File: resources/views/holidays/index.blade.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        <i class="mobis-i-plus"></i>
+            New Holiday
+        </a>
+    @endcomponent
+
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <div class="col-md-12 col-xs-12">
+
+        <table class="table table-hover table-bordered">
```
Comment (coreysan) [inline resources/views/holidays/index.blade.php]: No more tables for layouts. Talk to mwaka/see Mwaka's individuals or my branches index.php and branches _single.php for some hints.

File: routes/web.php [hunk 8]
```diff
@@ -25,6 +25,7 @@
     // TODO: still need to limit this to just one sacco
     Route::resource('saccos', 'SaccoController');
     Route::resource('users', 'UserController');
+    Route::resource('holidays','HolidayController');
```
Comment (coreysan) [inline routes/web.php]: Move to settings group

File: tests/Feature/HolidaysTest.php [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+     */
+     public function super_user_can_post_holiday()
+     {
+         $last_id = Holiday::all()->last()->id;
+
+         $holidays = $this->actingAs($this->superuser)
+                            ->post('/holidays',
+                            $this->holiday->getAttributes());
+
+        $this->assertFalse($last_id==Holiday::all()->last()->id);
```
Comment (coreysan) [inline tests/Feature/HolidaysTest.php]: beautiful :)

File: tests/Feature/HolidaysTest.php [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+          $response = $this->actingAs($this->janitor)
+                ->post('/holidays',
+                $this->holiday->getAttributes() );
+         $after =Holiday::all()->last()->id;
+
+         $this->assertTrue($before==$after);
+          //$response->assertStatus(403);
+               
+      }
+
```
Comment (coreysan) [inline tests/Feature/HolidaysTest.php]: Also test Delete method?

File: app/Api/saccoTranform.php [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?php
+/**
+ * Saccon transform
+ * 
+ * @param string $funcArg
```
Comment (FREDOCAKA) [inline app/Api/saccoTranform.php]: Deleted it

File: app/Http/Controllers/trial.php [hunk 12]
```diff
@@ -0,0 +1,84 @@
+<?php
```
Comment (FREDOCAKA) [inline app/Http/Controllers/trial.php]: just got rid of it

File: app/Http/Controllers/HolidayController.php [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+    {  
+        $holiday = Holiday::find($id)
+                  ->update([
+                     'name' => request('name'),
+                     'sacco_id' => Auth::user()->sacco->id,
+                     'date' => date('Y-m-d',strtotime(request('date'))),
+                    ]);
+
+        Flash::success('Holiday\'s update was successfull' );
+        return redirect('/holidays');
```
Comment (FREDOCAKA) [inline app/Http/Controllers/HolidayController.php]: Updated to '/settings/holidays/..
But in the navigation menu, the url is '**/settings/ holiday-settings/'.**, which has
to be changed to '/settings/holidays/..

File: app/Models/Holiday.php [hunk 14]
```diff
@@ -0,0 +1,37 @@
+<?php
```
Comment (FREDOCAKA) [inline app/Models/Holiday.php]: Thanks

File: database/factories/HolidayFactory.php [hunk 15]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+use App\Models\Holiday;
+
+/** @var \Illuminate\Database\Eloquent\Factory $factory */
+$factory->define(Holiday::class, function (Faker\Generator $faker) {
+    return [
+        'name' => $faker->name,
+        'sacco_id' => $faker->sacco_id,
```
Comment (FREDOCAKA) [inline database/factories/HolidayFactory.php]: use numberBetween(1,3)

File: database/migrations/2017_03_09_085400_create_holidays_table.php (hunk 2 above)
Comment (FREDOCAKA) [inline database/migrations/2017_03_09_085400_create_holidays_table.php]: Thanks

File: app/Http/Controllers/HolidayController.php (hunk 13 above)
Comment (coreysan) [inline app/Http/Controllers/HolidayController.php]: You change that, search the codebase for it and change it as needed

File: database/factories/HolidayFactory.php (hunk 15 above)
Comment (coreysan) [inline database/factories/HolidayFactory.php]: That only works if 3 saccos exist in the system! And what if 100 exist? Only the first 3 get holidays? Use this: 

`Sacco::inRandomOrder()->first()`

File: app/Scopes/Models/Customers/Customer.php [hunk 16]
```diff
@@ -0,0 +1,65 @@
+<?php
```
Comment (coreysan) [inline app/Scopes/Models/Customers/Customer.php]: You moved a bunch of files into the Scopes folder! 

Move them out/ensure they exist where they should. 

Also, scroll through your pull request to double-check it

**Pull request merged**